### PR TITLE
0.6.1

### PR DESCRIPTION
**Pull request by Ngoguey42**
---

### Pull request information

---

Welcome to PR number 42!!!. Don't look at the branch's name, this is the version `0.6.1`, not `6.1.0`.

As usual, please review the diffs and the release note below.

---

### Beginning of the anticipated release note for `0.6.1`

---

# Public changes
### Footprint coordinates rounding after reprojection
Thanks to the new parameter of the `Footprint.move` method, opening a raster when using an `sr_work` in the `Dataset` constructor work much better than before. 

### Parameters update
- Add `Footprint.move@round_coordinates`

# Private changes
- Unit tests now work with `GDAL 3.0`
- Add strong unit tests for `Footprint.move`

### CODE REVIEW

File: buzzard/_footprint.py [hunk 1]
```diff
@@ -453,6 +467,12 @@ def move(self, tl, tr=None, br=None):
             if br is not None:
                 raise ValueError('If br present, bl should be present too') # pragma: no cover
 
+        if round_coordinates:
+            if br is None: # pragma: no cover
+                raise ValueError(
+                    'Can only round input coordinates when all parameters are probided'
```
Comment (azarz) [inline buzzard/_footprint.py]:
```suggestion
                    'Can only round input coordinates when all parameters are provided'
```

File: buzzard/_footprint.py [hunk 2]
```diff
@@ -432,6 +433,19 @@ def move(self, tl, tr=None, br=None):
             New top right coordinates
         br: (nbr, nbr)
             New bottom right coordinates
+        round_coordinates: bool
+            Round the input coordinates with respect to `buzz.env.significant`, so that the output
+            `Footprint` is as much similar as possible as the input `Footprint` regarding those
```
Comment (Toz3wm) [inline buzzard/_footprint.py]: I would add something about cancelling the noise in the transformation matrix

File: buzzard/_footprint.py [hunk 3]
```diff
@@ -412,7 +413,7 @@ def _classify(obj):
             footprints, geoms, resolution, rotation, alignment
         )
 
-    def move(self, tl, tr=None, br=None):
+    def move(self, tl, tr=None, br=None, round_coordinates=False):
```
Comment (Toz3wm) [inline buzzard/_footprint.py]: If default value is set to `False`, soundn't error msg when creating complex footprint be updated with this new feature as well ?

File: buzzard/_footprint.py [hunk 4]
```diff
@@ -453,6 +467,12 @@ def move(self, tl, tr=None, br=None):
             if br is not None:
                 raise ValueError('If br present, bl should be present too') # pragma: no cover
 
+        if round_coordinates:
+            if br is None: # pragma: no cover
+                raise ValueError(
```
[the remark this answers is not in the document]
Comment (Ngoguey42) [reply buzzard/_footprint.py]: Added to TODO

File: buzzard/_footprint.py (hunk 3 above)
Comment (Ngoguey42) [reply buzzard/_footprint.py]: Done

File: buzzard/_footprint.py (hunk 2 above)
Comment (Ngoguey42) [reply buzzard/_footprint.py]: Done

**Pull request merged**